src/System.Buffers/src/System/Buffers/DefaultArrayPool.cs

```diff
@@ -20,26 +22,35 @@ internal DefaultArrayPool(int maxLength, int arraysPerBucket)
             if (arraysPerBucket <= 0)
                 throw new ArgumentOutOfRangeException("arraysPerBucket");
 
+            _eventSource = new ArrayPoolEventSource();
+
             // Our bucketing algorithm has a minimum length of 16
             if (maxLength < MinimiumArraySize)
                 maxLength = MinimiumArraySize;
             
             int maxBuckets = Utilities.SelectBucketIndex(maxLength);
-            _buckets = new ArrayPoolBucket<T>[maxBuckets + 1];
+            _buckets = new DefaultArrayPoolBucket<T>[maxBuckets + 1];
             for (int i = 0; i < _buckets.Length; i++)
-                _buckets[i] = new ArrayPoolBucket<T>(Utilities.GetMaxSizeForBucket(i), arraysPerBucket);
+                _buckets[i] = new DefaultArrayPoolBucket<T>(Utilities.GetMaxSizeForBucket(i), arraysPerBucket, _eventSource);
         }
 
+        public override EventSource TraceEventSource
```

Comment:
It doesn't need to be public.  It'll have a name/Guid that listeners can use to register for it.  We also don't want other arbitrary code calling these event logging methods, which are only relevant to the things they're tracing.  If someone else implements their own pool, then they can also implement their own EventSource for their pool.